src/Swarm/Game/Scenario/Cell.hs

```diff
@@ -22,13 +23,19 @@ import Swarm.Util.Yaml
 
 -- | A single cell in a world map, which contains a terrain value,
 --   and optionally an entity and robot.
-data Cell = Cell
+--   It is parameterized on the Entity type to facilitate less
+--   stateful versions of the Entity type in rendering scenario data.
+data PCell e = Cell
   { cellTerrain :: TerrainType
-  , cellEntity :: Maybe Entity
+  , cellEntity :: Maybe e
```

Comment:
So the idea is that instead of writing something like:
```Haskell
worldMapToText :: WorldPalette -> [[Cell]] -> Text
```
we would instead do it in two steps:
```
simplifyEntityForMe :: Entity -> SimplerEntity
simpleWorldMapToText :: WorldPalette (PCell SimplerEntity) -> Text
```
Is this because you want to have `ToJSON` (not `ToJSONE`) instance? 🤔

Reply:
@kostmo if this is about `ToJSON` we discussed with @byorgey, that you could simply use a custom tuple:
```Haskell
newtype Env e v = Env (e, v)
instance ToJSON (Env E Val) -- maybe we could codify this pattern as ToJSONE => ToJSON (Env, Val)
```
which is the same as
```Haskell
simplify :: E -> Val -> SimpleVal
instance ToJSON SimpleVal
```

Reply:
> Is this because you want to have ToJSON (not ToJSONE) instance?

I haven't quite wrapped my head around how `FromJSONE` works (or how `ToJSONE` would).

For the purpose of the World Editor, we just need to emit a fragment of the scenario file (either to a file or via the web API ).

I'm not sure how easy it is to read in its current state, but [here's what I'm trying to do in the World Editor PR](https://github.com/swarm-game/swarm/pull/873/files#r1048985734).

Reply:
After reading through your PR, I think it makes sense to keep it simple. 👍 

The idea with `FromJSONE` is that you need the environment to choose between parsing alternatives.

But with `ToJSON` no tricks are necessary, so I was incorrect in thinking that we would ever need it.